test/enigma_test.rb

```diff
@@ -147,30 +157,50 @@ def test_crack_encrypted_message_witout_date
     @enigma.stubs(:todays_date).returns("190420")
     expected =
     {
-      decryption: "this is an 'very' important message.
-      it needs to be encrypted!!! end",
+      decryption: "this is an 'very' important message.\nit needs to be encrypted!!! end",
       date: "190420",
       key: "14140"
     }
-
-    assert_equal expected, @enigma.crack("sjuwzkdd pl'ugcb'buqoqcx pedlgdw iq.
-    kedmgqhrbeszdqddpovxreic!!!zgzh")
+    assert_equal expected, @enigma.crack("mzwet fmuen'owek'rwzifefuegmfwfeuys.\n gmgwsqlrgattsmyeqdrggrx!!!twaq")
   end
 
   def test_find_key_with_date_and_encrypted_message
     assert_equal "08304", @enigma.find_key("vjqtbeaweqihssi", "291018")
-    
+
     assert_equal "14140", @enigma.find_key("mzwet fmuen'owek'rwzifefuegmfwfeuys.
      gmgwsqlrgattsmyeqdrggrx!!!twaq", "190420")
-
     assert_equal "20258", @enigma.find_key("sjuwzkdd pl'ugcb'buqoqcx pedlgdw iq.
     kedmgqhrbeszdqddpovxreic!!!zgzh", "190420")
   end
 
+  def test_find_first_key
+    assert_equal "08", @enigma.find_first_key(14, 6)
+    assert_equal "20", @enigma.find_first_key(26, 6)
+  end
+
+  def test_find_next_key_value
+    key = "08"
+    @enigma.find_next_key_value(key, 1, 3, 5)
+    assert_equal "083", key
```

Comment:
I may need to make these tests more robust to find out why the sometimes it finds the wrong key